pkg/registry/generic/registry/store.go

```diff
@@ -816,12 +790,6 @@ func (e *Store) filterAndDecorateFunction(m generic.Matcher) func(runtime.Object
 			glog.Errorf("unable to match watch: %v", err)
 			return false
 		}
-		if matches && e.Decorator != nil {
```

Comment:
this is the only place that decorator is coupled with filter today, right? Can we move this decorator out of the filter? We probably can call this outside the filter?